R/carehomes_rt.R

```diff
@@ -224,6 +224,199 @@ carehomes_Rt <- function(step, S, p, prob_strain = NULL,
+        draw_from(1, p$k_C_1, p$gamma_C_1)
+      infectivity <- 1
+    }
+    n_secondary_cases <- rpois(1, lambda = infectivity * sample_I)
+    tmp_GT <- seq(sample_E + p$dt, sample_E + sample_I, p$dt)
+    sample_GT <- c(sample_GT,
+                   sample(tmp_GT, n_secondary_cases, replace = TRUE))
+    i <- length(sample_GT)
+  }
+
+  sample_GT
+
+}
+
+##' @importFrom graphics hist
+draw_one_GT_distr <- function(p, n = 1000) {
+  sample_GT <- draw_one_GT_sample(p, n)
+  ## the discretisation below allows having a zero on the first day
+  ## which is required by EpiEstim
+  ## this is because the GT is computed from the sum of E and I
+  ## and each of those is at least 0.25 days
+  ## hence the GT is at least 0.5 day
+  hist(sample_GT, breaks = seq(0, ceiling(max(sample_GT) + 1), 1) - 0.51,
+         plot = FALSE)$density
+}
+
+
+## We expect 'pars' to be a list along sample
+## We expect 'step' to be a vector along step
+
+##' Compute "Rt" using EpiEtim for a set of simulated trajectories (e.g., the
+##' result of the `$iterate()` method of [carehomes], [mcstate::pmcmc()] or
+##' [mcstate::pmcmc_predict()]. The trajectories should share
+##' parameters.
+##'
+##' @title Compute Rt using EpiEstim for a set of trajectories
+##'
+##' @param step A vector of steps
+##'
+##' @param incidence A matrix (n trajectories x n steps) of incidence counts
+##'
+##' @param p A single [carehomes_parameters()] object.
+##'
+##' @param sliding_window_ndays An integer giving the length of the sliding
+##' window on which Rt will be estimated
+##'
+##' @param mean_prior The mean prior for Rt
+##'
+##' @param sd_prior The standard deviation of the prior for Rt
+##'
+##' @param n_GT An integer giving the number of generation times to be drawn to
+##' construct the discrete distribution of the generation time
+##'
+##' @param n_R An integer giving the number of Rt values to sample from for each
+##' incidence trajectory. These will then be aggregated across all incidence
+##' trajectories.
+##'
+##' @param save_all A boolean determining whether to save all samples of Rt
+##' estimated or only a summary
+##'
+##' @return A list with elements
+##' `t_start` (vector of first days of the sliding windows over which Rt is
+##' estimated),
+##' `t_end` (vector of last days of the sliding windows over which Rt is
+##' estimated),
+##' `Rt` a matrix (only present if `save_all = TRUE`) containing for each
+##' sliding window (each row in the matrix) a sample of n_R * nrow(inc) values
+##' of Rt for that sliding window (columns of the matrix)
+##' `Rt_summary` a matrix containing for each sliding window (each row in the
+##' matrix) the 2.5%, 50%, 97.5% quantiles and the mean of Rt for that sliding
+##' window (columns of the matrix)
+##'
+##' @export
+##' @importFrom EpiEstim make_config estimate_R
+##' @importFrom epitrix gamma_mucv2shapescale
+##' @importFrom stats quantile rgamma
+carehomes_EpiEstim_Rt_trajectories <- function(step, incidence, p,
```

Comment:
I think `carehomes_Rt_trajectories_EpiEstim` would be marginally better. However, there's no great name here, and the casing here irritates me a lot. I would personally prefer `carehomes_rt_trajectories_epiestim` with the package renamed